book/forms.md

```diff
@@ -382,69 +367,56 @@ def submit_form(self, elt):
     self.load(url, body)
 ```
 
-This adds a new parameter to the browser's `load` method, which we can
-pass to `request`:
+This uses a new parameter to the browser's `load` method for the
+request body. Let's pass that through to `request`:
 
-``` {.python}
+``` {.python indent=4}
 def load(self, url, body=None):
     # ...
-    header, body = request(url, body)
+    headers, body = request(url, body)
 ```
 
-Sixth and finally, to actually send a POST request, we need to modify
-the `request` function to support the new argument. First, the method
-needs to be configurable:
+Then `request` can send the that request body. That requires a few
+changes to `request`. First, it needs to use `POST`, not `GET`:
 
 ``` {.python}
 def request(url, payload=None):
     # ...
     method = "POST" if payload else "GET"
+    # ...
+    body = "{} {} HTTP/1.0\r\n".format(method, path)
 ```
 
-We need to use this method:
+Then we need to send the `Content-Length` header, which is mandatory
+on `POST` requests:
 
 ``` {.python}
 def request(url, payload=None):
     # ...
-    body = "{} {} HTTP/1.0\r\n".format(method, path)
-    body += "Host: {}\r\n".format(host)
-    body += "\r\n" + (payload or "")
-    s.send(body.encode("utf8"))
+    if payload:
+        length = len(payload.encode("utf8"))
+        body += "Content-Length: {}\r\n".format(length)
+    # ...
 ```
 
-Also, when you send a payload like this in the request, you need to
```

Comment:
It seems the diff is larger than this PR actually contains. The text here is not actually removed by your chain of edits, right?

Reply:
Looking this over—I think this text is removed because the `Content-Length` header is described (and described as mandatory) in the first "How forms work" section now. You think we should review here? I could see that.

Reply:
Ok that part makes sense, but what about the part that used to be there about the "enter" key?

Reply:
I removed the Enter key submit handling—it's now a little more involved than before thanks to the Browser/Tab split. Also, I think it would make a good exercise.

Reply:
Added the exercise